tsdb/engine.go

```diff
@@ -186,9 +186,8 @@ type EngineOptions struct {
 	// nil will allow all combinations to pass.
 	ShardFilter func(database, rp string, id uint64) bool
 
-	Config         Config
-	SeriesIDSets   SeriesIDSets
-	FieldValidator FieldValidator
```

Comment:
Eliminating Engine.FieldValidator seems reasonable, since I can't find any use other than the default validator. If we get rid of Engine.FieldValidator, should we also get rid of EngineOptions.FieldValidator? There might be confusion in the future if someone thinks they can set that to change the field validation behavior.

Reply:
This is in `EngineOptions` - I don't see any `FieldValidator` remaining in the code?

Reply:
I'm not sure why I thought there was an Engine.FieldValidator. I _thought_ there was a Shard.FieldValidator, typed Engine.FieldValidator, but it was really setting the value in EngineOptions.FieldValidator. My mistake, carry on!